cpp/vm/evmzero/interpreter.cc

```diff
@@ -976,6 +976,27 @@ struct Impl<OpCode::BASEFEE> {
   }
 };
 
+template <>
+struct Impl<OpCode::BLOBHASH> {
+  constexpr static OpInfo kInfo{
+      .pops = 1,
+      .pushes = 1,
+      .static_gas = 3,
+      .introduced_in = EVMC_CANCUN,
+  };
+
+  static OpResult Run(uint256_t* top, Context& ctx) noexcept {
+    const auto index = static_cast<int64_t>(top[0]);
```

Comment:
the check in line 991 may be too late, since you already rounded up the value in this line. Can you check if this is true? Other instructions may ask up-front if the stack value fits in an int64 before reading and casting it. 
I think you may need to pop it into a int256 variable first, and ask if it overflows.

Reply:
the check on line 991 that checks regarding `int64_t` max, checks against the "uncasted" value. I simply cast it here so that if the max check passes, I don't need to cast inside the if, and then again to use it as index.

Reply:
I see now, I  got confused, because you sometimes use index and or top[0] this could use less complication and be a little more straight forward.

Reply:
nice offline discussion on how to improve !!! thanks